src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        ///   Exposes methods to set goal time <see cref="Goal"/>
+        /// </summary>
+        /// <param name="ChannelName"></param>
+        /// <param name="DefinitionId"></param>
+        /// <param name="Minutes"></param>
+        [HttpPost]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.Created, Type = typeof(Models.Goal), Description = "Goal for given Channel and Definition-Id is created/updated")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([Required] int DefinitionId, [Required] string ChannelName, [Required] int Minutes)
```

Comment:
I think "minutes" should be [FromForm]. Accepting as a query parameter is... odd...

Reply:
I changed the name to GoalInMinutes. That is more meaningful than minutes. 

If I follow FromForm then its going to be something like RepoOwnerAddGoal, its pretty long.

Reply:
[FromForm] just tells ASP.NET (and therefore swagger, therefore autorest, therefore our generated libraries) where that value is supposed to be read from in the incoming request. The default is "everywhere", which is rarely desirable, and the client code will send it in a query parameter, which means it has an empty post body, which is weird.

The pose would go to `POST goals/1/.NETCore5?goalTimeInMinutes15`.  That's... awkward in REST land. Usually the thing you post is the body of the post request, and the GET request with the same URL/parameters should return that value.

Reply:
@ChadNedzlek 

I can probably can do this,by making it more descriptive 

goals/definitionId/2/channelName/.Net Core 5/goalInMinutes/3

what do you think about this?

Reply:
The goal is that 
`POST some/random/url` should have a mirror of `GET some/random/url`.  Something like [this (Wikipedia REST)](https://en.wikipedia.org/wiki/Representational_state_transfer#Relationship_between_URI_and_HTTP_methods).

The "goalTimeInMinutes" would make more sense in the body of the POST rather than in the URL (it's what you are **POST**ing... the new goal, the "definition" and "channel" are what you are posting to).  Think of the URL as the address on an envelope, and the body of the POST as the contents. The address is "definition 50 in the .NET Core Channel", what's in the envelop is "the goal is 50".  So if you ask for the contents of "the envelope that says definition 50 in the .NET Core Channel" it will GET you number "50" back.

By default, if you don't specify _where_ each parameter in a MVC method comes from, they are read from arbitrary places. [[FromForm]](https://docs.microsoft.com/en-us/aspnet/core/mvc/models/model-binding?view=aspnetcore-3.0#sources) makes this explicit as to where it's supposed to come from, and tells things that they should post them in the body of the request as form formatted data.

You could also use [FromBody], which would make it be a raw JSON number that gets sent, but that's a touch odd; usually one would expect to post a full JSON object as the payload, something like `{goalTime:"3:00.000"}`, which would be nice, but requires creating a new type that you accept as the body, which might be a bit heavy for a single value.

It might also be nice to accept a "timespan", so the internal fact that it's minutes doesn't have to be exposed, and we can accept strings that look like times to people (and if we want to change it to hours or seconds internally, this API can still function perfectly well). And you don't have to worry about people not reading the docs and sending "1" hour instead of "60" minutes, or "3600" seconds.

Reply:
I changed the url for the getGoal method

[HTTPGET]
url :/goals/channelName/.Net Core 5/definitionId/34

[HTTPPOST] : this posts the body to the db.
url : /goals 
body :
     {
        channelName: "string"
        definitionId : int 
        minutes : int 
      }

Since goal is going to be only in minutes, I am going to keep this as minutes.